elastic/kibana
# [SavedObjects] Create serverless utils for jest integration tests

## Summary

Introduce basic utils for instantiating serverless servers ;). Also adds a simple smoke test (that is currently skipped) demonstrating usage.

## How to test

Run docker locally, check this branch out locally, un`skip` the smoke test and run:

```
yarn test:jest_integration ./src/core/server/integration_tests/saved_objects/serverless/migrations
```

Note: the test cannot be unskipped as we are blocking on https://github.com/elastic/kibana/pull/162673 but we would like to prepare our tests so long --- they can be locally executed.

## Follow up

* Create Jest integration tests using these helpers for all migration actions (see `packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/actions` and `packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions`).
* Migrate our existing ZDT tests to run against serverless Elasticsearch

## Related

https://github.com/elastic/kibana/pull/162673
